docs/reference/modules/asset.md

```diff
@@ -0,0 +1,205 @@
+
+<AposCodeBlock>
+
+```javascript
+import  { customMethod } from ('Utilities/methods.js');
+
+export default () => {
+// Code utilizing customMethod
+};
+```
+<template v-slot:caption>
+modules/my-module/ui/src/index.js
+</template>
+</AposCodeBlock>
+
+Adding a unique name for the object within the apostrophe `extensions` property is important. Passing an object with the same name to an `extensions` property of another module will cause a conflict. The configuration file will be set to the `extensions` object in the last loaded module.
+
+#### `extensions` function
+
+Functions in the `extensions` property take an `option` parameter to allow multiple modules to modify the same extension through the [`extensionOptions`](#extensionoptions) property. Each property to be altered across multiple modules must be present in the option that was originally returned.
+
+<AposCodeBlock>
+
+```javascript
+const path = require("path");
+
+module.exports = {
+ // ...
+  webpack: {
+   extensions: {
+     addAlias(options) {
+       return {
+         resolve: {
+           extensions: [ ...(options.extensions || []), '.jsx' ],
+           alias: {
+             ...(options.alias || {}),
+             Utilities: path.resolve(process.cwd(), './lib/utils')
+           }
+         }
+       };
+     }
+   }
+ },
+ // ...
+};
+
+```
+<template v-slot:caption>
+modules/my-module/index.js
+</template>
+</AposCodeBlock>
+
+### `extensionOptions`
+Any module can add additional functionality to an existing webpack `extension` that was created through a function using the `extensionOptions` property. This property takes a function with an `options` parameter and returns an object with additional options for any configuration options present in the original function.
+
+#### Example
+
+Extending the previous example with an additional extension and alias:
+
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+ // ...
+  webpack: {
+   extensionOptions: {
+     addAlias(options) {
+       return {
+         extensions: [ '.jsm' ],
+         alias: {
+           Data: path.resolve(process.cwd(), './lib/data')
+         }
+       };
+     }
+   }
+ },
+// …
+};
+```
+<template v-slot:caption>
+module/my-new-widget/index.js
+</template>
+</AposCodeBlock>
+
+### `bundles`
+The `bundles` option allows delivery of frontend assets to a specific page or if a widget is on a page page. This option takes an object containing either the JavaScript or stylesheet file names without extension as properties. These files should be located in the module's `ui/src` folder. Each property takes either an empty object or a sub-property of `templates`. The `templates` sub-property accepts an array of page template names where the content should be loaded. So, for example, a bundle added to a `piece-page-type` could specify `templates: [ 'index' ]` to load the bundle on the `index` page, but not the `show` page.
```

Comment:
The `bundles` subproperty can be present in any type of module, not just widget type and page type modules. However, configuring a bundle in a widget type or page type module is the only way to cause it to actually be loaded. If a bundle is present in another type of module, that module can contribute code to it, but it will only get loaded when also configured for at least one widget type or page type. The `templates` feature is most often relevant for piece-page-type modules; if it is not used, it is assumed both `index` and `show` templates should load the bundle.

Reply:
Changed.